packages/x-data-grid-premium/src/hooks/features/rowGrouping/useGridDataSourceRowGroupingPreProcessors.ts

```diff
@@ -0,0 +1,139 @@
+      if (!getChildrenCount) {
+        throw new Error('MUI X: No `getChildrenCount` method provided with the dataSource.');
+      }
+
+      const sanitizedRowGroupingModel = gridRowGroupingSanitizedModelSelector(apiRef);
+      const columnsLookup = gridColumnLookupSelector(apiRef);
+      const groupingRules = getGroupingRules({
+        sanitizedRowGroupingModel,
+        columnsLookup,
+      });
+      apiRef.current.caches.rowGrouping.rulesOnLastRowTreeCreation = groupingRules;
+
+      const getRowTreeBuilderNode = (rowId: GridRowId) => {
+        const parentPath = (params.updates as GridRowsPartialUpdates).groupKeys ?? [];
+        const row = params.dataRowIdToModelLookup[rowId];
+        const groupingRule = groupingRules[parentPath.length];
+        const groupingValueGetter = groupingRule?.groupingValueGetter;
+        const leafKey =
+          groupingValueGetter?.(
+            row[groupingRule.field] as never,
+            row,
+            columnsLookup[groupingRule.field],
+            apiRef,
+          ) ?? getGroupKey(params.dataRowIdToModelLookup[rowId]);
+        return {
+          id: rowId,
+          path: [...parentPath, leafKey ?? rowId.toString()].map((key, i) => ({
+            key,
+            field: groupingRules[i]?.field ?? null,
+          })),
+          serverChildrenCount: getChildrenCount(params.dataRowIdToModelLookup[rowId]) ?? 0,
+        };
+      };
+
+      if (params.updates.type === 'full') {
+        return createRowTree({
+          previousTree: params.previousTree,
+          nodes: params.updates.rows.map(getRowTreeBuilderNode),
+          defaultGroupingExpansionDepth: props.defaultGroupingExpansionDepth,
+          isGroupExpandedByDefault: props.isGroupExpandedByDefault,
+          groupingName: ROW_GROUPING_STRATEGY_DATA_SOURCE,
+        });
+      }
+
+      return updateRowTree({
+        nodes: {
+          inserted: (params.updates as GridRowsPartialUpdates).actions.insert.map(
+            getRowTreeBuilderNode,
+          ),
+          modified: (params.updates as GridRowsPartialUpdates).actions.modify.map(
+            getRowTreeBuilderNode,
+          ),
+          removed: (params.updates as GridRowsPartialUpdates).actions.remove,
+        },
+        previousTree: params.previousTree!,
+        previousGroupsToFetch: params.previousGroupsToFetch,
+        previousTreeDepth: params.previousTreeDepths!,
+        defaultGroupingExpansionDepth: props.defaultGroupingExpansionDepth,
+        isGroupExpandedByDefault: props.isGroupExpandedByDefault,
+        groupingName: ROW_GROUPING_STRATEGY_DATA_SOURCE,
+      });
+    },
+    [
+      apiRef,
+      props.unstable_dataSource,
+      props.defaultGroupingExpansionDepth,
+      props.isGroupExpandedByDefault,
+    ],
+  );
+
+  const filterRows = React.useCallback<GridStrategyProcessor<'filtering'>>(() => {
+    const rowTree = gridRowTreeSelector(apiRef);
+
+    return skipFiltering(rowTree);
```

Comment:
what is preventing you reseting the grid and making a new request with the new parameters?

Reply:
Not sure if I follow 🤔. This is the client-side strategy processor for filtering. Since we don't have client side filtering with the `dataSource`, we skip it.

Just like for tree data:
https://github.com/mui/mui-x/blob/46df9666a5d629548f68f5fe3f6c10b7a31248b1/packages/x-data-grid-pro/src/hooks/features/serverSideTreeData/useGridDataSourceTreeDataPreProcessors.tsx#L200-L210

Did I get your point right?

Reply:
ah yes, ignore this
wrong line
I was looking at the point about the group column sorting and actually wanted to comment this on the group cell definition, but ended up here 😅
I will resolve this and make the same comment there